scantool/diag_l2.c

```diff
@@ -57,7 +57,7 @@ static struct {
 } l2internal = {
 	.dl2conn_list = NULL,
 	.dl2l_list = NULL,	// linked-list of current L2-L0 links
-	.init_done = NULL
+	.init_done = false
```

Comment:
indeed, but if you find other unrelated things like this they should be separate commits

Reply:
This was the only not type conversion related warning in level 3 - so I changed it.